### ISSUE TITLE
BuildWatcher starts to observe incorrect build

### ISSUE THREAD

**Issue by allu22** [fastlane/fastlane#10750]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
We uploaded a new build to a new train (with `skip_waiting_for_build_processing` option set to `false`), while we had a build in previous train waiting for a (TestFlight) review. But because of parse error, fastlane ended up observing build that was waiting for a review.

When `BuildWatcher.watching_build` fails to get processing build from `Spaceship::TestFlight::Build.all_processing_builds` it tries to fall back to latest build it can find. https://github.com/fastlane/fastlane/blob/a4704d7a0da18e35a24d858000f66e80c8956ea1/fastlane_core/lib/fastlane_core/build_watcher.rb#L29

But if there is a problem fetching builds for a new train, it selects newest build from previous train. 
In our case, where that build was waiting for a review, it started to wait for a build to be reviewed.
https://github.com/fastlane/fastlane/blob/a4704d7a0da18e35a24d858000f66e80c8956ea1/fastlane_core/lib/fastlane_core/build_watcher.rb#L15

Only way this loop could now be broken is if the review passed successfully and the external state for this build became active.
https://github.com/fastlane/fastlane/blob/a4704d7a0da18e35a24d858000f66e80c8956ea1/spaceship/lib/spaceship/test_flight/build.rb#L133-L135

But there is very likely state that is not considered, `testflight.build.state.review.rejected`. If that build is rejected in review, then the loop is never broken.

### Possible fixes
  - Remove fallback to latest build.  
https://github.com/fastlane/fastlane/blob/a4704d7a0da18e35a24d858000f66e80c8956ea1/fastlane_core/lib/fastlane_core/build_watcher.rb#L29  
  I am not sure what the reasoning is for this fallback. Only reason I could come up is if the build was processed immediately.

  - Add rejected to `processed?` definitions.  
https://github.com/fastlane/fastlane/blob/a4704d7a0da18e35a24d858000f66e80c8956ea1/spaceship/lib/spaceship/test_flight/build.rb#L133-L135
  This will break the loop but `BuildManager` will continue with incorrect build.

  - Determine build train and build from ipa container file that will be uploaded / was uploaded.  
  As `BuildWatcher` needs train version and build version to monitor the status of uploaded build, we could read them from ipa container that was uploaded by `BuildManager`.
  This solution makes an assumption that `CFBundleVersion` will become `build_version` and `CFBundleShortVersionString` becomes `train_version`. But I believe these are safe assumptions.

**Comment by ohayon** [comment]
Hey @allu22 - thanks for such a thorough report! Yeah we had set it to default to latest build because we thought that made sense at the time but I dont know if we considered this use case. Would you want to submit a PR for this fix (I think I would prefer solution 2 or 3 that you mentioned above as I think we should still fall back to the latest build if one isnt found. 🚀

**Comment by allu22** [comment]
@ohayon, I've submitted a PR for 3rd solution: https://github.com/fastlane/fastlane/pull/10766

**Comment by luketomlinson** [comment]
I'm having a similar issue. I think iTunes connect has changed a little bit. Newly uploaded builds don't seem to appear in the list right away(builds used to show up and just be processing, but now there appears to be some latency). Because the build watcher only grabs the latest, `pilot` always try to release the latest build to testers, when it should be waiting on a new build to process.

**Comment by agordeev** [comment]
I'm getting the same error with `2.68.2`:

[17:26:30]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
[17:26:49]: Started watching build 1.1.3 - 58 but expected 1.1.4 - 60
[17:26:53]: Successfully finished processing the build 1.1.3 - 58

**Comment by ohayon** [comment]
Hey @agordeev, would you mind sharing the entire output from your terminal as well as the output from `fastlane env` so we can see the full context? Thanks! 🚀

**Comment by coverback** [comment]
@ohayon I have the same issue since couple days ago.
Here's my output (although then Jenkins calls it, there are some more env variables, which don't affect the outcome).

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                                              |
| --------------------------- | ---------------------------------------------------------------------------------- |
| OS                          | 10.13.1                                                                            |
| Ruby                        | 2.4.2                                                                              |
| Bundler?                    | true                                                                               |
| Git                         | git version 2.13.6 (Apple Git-96)                                                  |
| Installation Source         | /path/jenkins_workspace/deploy_testflight/vendor/bundle/ruby/2.4.0/bin/fastlane    |
| Host                        | Mac OS X 10.13.1 (17B1003)                                                         |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.4.2_1/lib                                                 |
| OpenSSL Version             | OpenSSL 1.0.2l  25 May 2017                                                        |
| Is contained                | false                                                                              |
| Is homebrew                 | false                                                                              |
| Is installed via Fabric.app | false                                                                              |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                                        |
| Xcode Version               | 9.1                                                                                |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

$:.unshift File.dirname(__FILE__)
require 'git_helper.rb'
require 'helpers.rb'

VERSION_NUMBER_REGEX = /^[0-9]\.[0-9]{1,3}(\.[0-9]{1,3})?$/

default_platform :ios

platform :ios do

  before_all do |lane, options|
    load_shared_variables
    setup_jenkins(unlock_keychain: options[:sign] || false,
                  keychain_password: options[:pw])
    reset_git if is_ci?
  end

  # Public lanes

  desc "Upload and download new localization resorces to/from Crowdin"
  lane :update_localization do
    run_localization_scripts
    create_pr_from_current_changes(pr_title: "Update strings from Crowdin")
  end

  desc "Run all tests that are included into your project"
  lane :test do
    run_tests(skip_slack: true)
    generate_test_report

    coverage if !is_ci?

    lint
  end

  desc "Run all local pod tests that are included into your project"
  lane :test_local_pods do
    run_local_pod_tests
  end

  desc "Bump default version for beta build"
  lane :bump_version do |options|
    unless options[:version] && options[:version].match(VERSION_NUMBER_REGEX)
      UI.user_error!("No version specified. Use 'version:7.4' syntax")
    end

    increment_version_number(version_number: options[:version])
    commit_version_bump(message: "Bump version to v#{options[:version]}")
    push_to_git_remote if is_ci?
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  lane :beta do
    prepare_env
    sync_code_signing(type: "appstore", readonly: true)
    build_app
    commit_log = recent_commit_log(from_commit: ENV['GIT_PREVIOUS_SUCCESSFUL_COMMIT'])
    testflight(changelog: commit_log)
    upload_dsym
    success_release_slack_notification(app_name: "Pipedrive β", recent_commit_log: commit_log)
  end

  desc "Submit a new Release Build to Apple TestFlight"
  lane :release do
    prepare_env
    sync_code_signing(type: "appstore", readonly: true)
    build_app
    before_tagging_commit_log = recent_commit_log(from_commit: last_git_tag)
    testflight(changelog: before_tagging_commit_log,
               skip_submission: true)
    git_tag_release
    upload_dsym
    success_release_slack_notification(app_name: "Pipedrive",
                                       recent_commit_log: before_tagging_commit_log)
  end

  # Complex lane implementations

  def run_local_pod_tests
    local_pods = "LocalPods/"
    example = "Example/"
    local_pods_dir = "#{parent_directory}/#{local_pods}"
    local_pods_folders = directories_in_directory(directory: local_pods_dir)

    local_pods_folders.each { |pod|
      pod_example_dir = "#{local_pods_dir}#{pod}/#{example}"

      if directory_exists(directory: pod_example_dir)
        podfile_dir = "#{local_pods}#{pod}/#{example}"
        cocoapods(podfile: podfile_dir, try_repo_update_on_error: true)

        pod_workspace_path = "#{podfile_dir}#{pod}.xcworkspace"
        pod_scheme_name = "#{pod}-Example"
        run_tests(workspace: pod_workspace_path, scheme: pod_scheme_name, skip_slack: true)
      end
    }
  end

  def run_localization_scripts
    parent_directory = File.expand_path("..", Dir.pwd)
    Dir.chdir parent_directory
    sh("./scripts/export_strings_to_crowdin.sh . #{ENV["CROWDIN_API_TOKEN"]}")
    sh("./scripts/import_strings_from_crowdin.sh . #{ENV["CROWDIN_API_TOKEN"]}")
  end

  def create_pr_from_current_changes(pr_title:)
    status = sh("git status --porcelain")
    if !status.nil? && !status.empty?
      current_time = DateTime.now.strftime "%d-%m-%Y-%H-%M-%S"
      branch_name = "pr-#{current_time}"
      sh("git checkout -b #{branch_name}")
      sh("git commit -a -m \"#{pr_title}\"")
      sh("git push origin #{branch_name}")
      create_pull_request(repo: ENV["GITHUB_PROJECT_NAME"],
                          title: pr_title,
                          head: branch_name)
    else
       UI.message("No changes")
    end
  end

  def git_tag_release
    version = lane_context[SharedValues::VERSION_NUMBER]
    build_number = lane_context[SharedValues::BUILD_NUMBER]
    desired_tag = "v#{version}-#{build_number}"

    if current_git_tags.include?(desired_tag)
      puts "HEAD is already tagged with #{possible_existing_tag}, skipping."
    else
      add_git_tag(tag: desired_tag)
      push_git_tags
    end
  end

  def upload_dsym
    bundle_identifier = CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier)

    version = Actions.lane_context[SharedValues::VERSION_NUMBER]
    build_number = Actions.lane_context[SharedValues::BUILD_NUMBER]

    puts "Start download dSYM for bundle_identifier - #{bundle_identifier}   version - #{version}(#{build_number})"
    download_dsyms(version: version,
                   build_number: build_number)

    absolute_old_dsym_path = "#{parent_directory}/#{bundle_identifier}-#{version}-#{build_number}.dSYM.zip"
    upload_symbols_to_crashlytics(dsym_path: absolute_old_dsym_path)

    absolute_new_dsym_path = Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH]
    puts "Move downloaded dSYM file FROM #{absolute_old_dsym_path} TO #{absolute_new_dsym_path}"
    sh("mv #{absolute_old_dsym_path} #{absolute_new_dsym_path}")
  end

  def success_release_slack_notification(app_name:, recent_commit_log:)
    payload = {
      "Build version" => Actions.lane_context[SharedValues::VERSION_NUMBER],
      "Build number" => Actions.lane_context[SharedValues::BUILD_NUMBER],
      "Commit log" => linkable_commit_log(log: recent_commit_log),
      "Disabled prod features" => disabled_features
    }
    slack(
      message: "#{app_name} successfully released!",
      success: true,
      payload: payload.reject { |k,v| v.nil? },
      default_payloads: []
    )
  end

  def lint
    output_file = "fastlane/swiftlint/output/report"
    config_file = "fastlane/swiftlint/configs/swiftlint.yml"
    swiftlint(
      output_file: "#{output_file}.xml",
      config_file: config_file,
      ignore_exit_status: true,
      reporter: "junit",
      quiet: true
    )
  end

  def coverage
    slather(html: true)
    fix_slather_js
  end

  def prepare_env
    ensure_version_is_known
    set_version_in_context
  end

  def generate_test_report
    output_directory = ENV['FL_SETUP_JENKINS_OUTPUT_DIRECTORY']
    result = trainer(output_directory: output_directory)

    if result.keys.count > 0
      generated_file_name = "../#{result.keys[0]}"
      File.rename(generated_file_name, "../#{output_directory}/test_report.xml")
    end
  end

  # Fastlane-dependant helpers

  def ensure_version_is_known
    unless version_set_in_environment? || git_head_is_version_tagged?
      UI.user_error!('No version info found! Pass "JENKINS_BUILD_NUMBER" env variable or set a git tag.')
    end
  end

  def set_version_in_context
    if version_set_in_environment?
      set_version_from_environment
    else
      parse_git_tag
    end

    UI.success("Building with version #{lane_context[SharedValues::VERSION_NUMBER]}"\
               "-#{lane_context[SharedValues::BUILD_NUMBER]}")
  end

  def set_version_from_environment
    lane_context[SharedValues::VERSION_NUMBER] = get_version_number
    lane_context[SharedValues::BUILD_NUMBER] = ENV['JENKINS_BUILD_NUMBER']

    # This is used when building, in Xcode
    ENV['LANE_BUILD_NUMBER'] = lane_context[SharedValues::BUILD_NUMBER]
  end

  def load_shared_variables
    begin
      import("keys")
    rescue => exception
      UI.error("fastlane/keys file not found")
      raise exception
    end
  end

  def parse_git_tag
    current_git_tags.each do |tag|
      if tag.match(VERSION_REGEX)
        lane_context[SharedValues::VERSION_NUMBER] = version_from_git_tag(tag)
        lane_context[SharedValues::BUILD_NUMBER] = build_number_from_git_tag(tag).to_s
        return
      end
    end
  end

end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
apple_id "mail@host.com"
team_id "ABC978YUDD"

for_lane :beta do
  app_identifier "com.pipedrive.beta.Pipedrive"
end

for_lane :release do
  app_identifier "com.pipedrive.Pipedrive"
end

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.68.2  | ✅ Up-To-Date  |


### Loaded fastlane plugins:

| Plugin                  | Version | Update-Status |
| ----------------------- | ------- | ------------- |
| fastlane-plugin-trainer | 0.4.0   | ✅ Up-To-Date  |


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| bundler                   | 1.16.0       |
| io-console                | 0.4.6        |
| CFPropertyList            | 2.3.5        |
| concurrent-ruby           | 1.0.5        |
| i18n                      | 0.9.1        |
| minitest                  | 5.10.3       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.4        |
| activesupport             | 4.2.10       |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| clamp                     | 0.6.5        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.3.1        |
| cocoapods-deintegrate     | 1.0.1        |
| cocoapods-downloader      | 1.1.3        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.11.0       |
| cocoapods-trunk           | 1.3.0        |
| cocoapods-try             | 1.1.0        |
| colored2                  | 3.1.2        |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.0.3        |
| molinillo                 | 0.5.7        |
| ruby-macho                | 1.1.0        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.3        |
| cocoapods                 | 1.3.1        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.1        |
| excon                     | 0.59.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.13.1       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.0        |
| jwt                       | 2.1.0        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.2       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| google-api-client         | 0.13.6       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.3.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.3        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 1.0.0        |
| trainer                   | 0.7.0        |
| fastlane-plugin-trainer   | 0.4.0        |
| mini_portile2             | 2.1.0        |
| nokogiri                  | 1.6.8.1      |
| slather                   | 2.4.3        |
</details>


*generated on:* **2017-12-11**
</details>



<details><summary>Lane log output</summary>

```
[deploy_testflight] $ /bin/sh -xe /var/folders/gd/wn10jrv57yn5lbznvb2y463c0000gn/T/jenkins6968986508250328161.sh
+ export LANG=en_US.UTF-8
+ LANG=en_US.UTF-8
+ export PATH=/Users/pipedrive/.fastlane/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
+ PATH=/Users/pipedrive/.fastlane/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
+ bundle install --path vendor/bundle
Using CFPropertyList 2.3.5
Using concurrent-ruby 1.0.5
Using i18n 0.9.1
Using minitest 5.10.3
Using thread_safe 0.3.6
Using tzinfo 1.2.4
Using activesupport 4.2.10
Using public_suffix 2.0.5
Using addressable 2.5.2
Using babosa 1.0.2
Using bundler 1.16.0
Using claide 1.0.2
Using clamp 0.6.5
Using fuzzy_match 2.0.4
Using nap 1.1.0
Using cocoapods-core 1.3.1
Using cocoapods-deintegrate 1.0.1
Using cocoapods-downloader 1.1.3
Using cocoapods-plugins 1.0.0
Using cocoapods-search 1.0.0
Using cocoapods-stats 1.0.0
Using netrc 0.11.0
Using cocoapods-trunk 1.3.0
Using cocoapods-try 1.1.0
Using colored2 3.1.2
Using escape 0.0.4
Using fourflusher 2.0.1
Using gh_inspector 1.0.3
Using molinillo 0.5.7
Using ruby-macho 1.1.0
Using nanaimo 0.2.3
Using xcodeproj 1.5.3
Using cocoapods 1.3.1
Using colored 1.2
Using highline 1.7.10
Using commander-fastlane 4.4.5
Using declarative 0.0.10
Using declarative-option 0.1.0
Using unf_ext 0.0.7.4
Using unf 0.1.4
Using domain_name 0.5.20170404
Using dotenv 2.2.1
Using excon 0.59.0
Using multipart-post 2.0.0
Using faraday 0.13.1
Using http-cookie 1.0.3
Using faraday-cookie_jar 0.0.6
Using faraday_middleware 0.12.2
Using fastimage 2.1.0
Using jwt 2.1.0
Using little-plugger 1.1.4
Using multi_json 1.12.2
Using logging 2.2.2
Using memoist 0.16.0
Using os 0.9.6
Using signet 0.8.1
Using googleauth 0.6.2
Using httpclient 2.8.3
Using mime-types-data 3.2016.0521
Using mime-types 3.1
Using uber 0.1.0
Using representable 3.0.4
Using retriable 3.1.1
Using google-api-client 0.13.6
Using json 2.1.0
Using mini_magick 4.5.1
Using multi_xml 0.6.0
Using plist 3.3.0
Using rubyzip 1.2.1
Using security 0.1.3
Using slack-notifier 1.5.1
Using terminal-notifier 1.8.0
Using unicode-display_width 1.3.0
Using terminal-table 1.8.0
Using tty-screen 0.6.3
Using word_wrap 1.0.0
Using rouge 2.0.7
Using xcpretty 0.2.8
Using xcpretty-travis-formatter 1.0.0
Using fastlane 2.68.2
Using trainer 0.7.0
Using fastlane-plugin-trainer 0.4.0
Using mini_portile2 2.1.0
Using nokogiri 1.6.8.1
Using slather 2.4.3
Bundle complete! 4 Gemfile dependencies, 85 gems now installed.
Bundled gems are installed into `./vendor/bundle`
+ bundle exec fastlane beta
+-------------------------+---------+---------+
|                Used plugins                 |
+-------------------------+---------+---------+
| Plugin                  | Version | Action  |
+-------------------------+---------+---------+
| fastlane-plugin-trainer | 0.4.0   | trainer |
+-------------------------+---------+---------+

[13:31:34]: ------------------------------
[13:31:34]: --- Step: default_platform ---
[13:31:34]: ------------------------------
[13:31:34]: Driving the lane 'ios beta' 🚀
[13:31:34]: ---------------------------
[13:31:34]: --- Step: setup_jenkins ---
[13:31:34]: ---------------------------

+-----------------------------+----------------------+
|          Summary for Setup Jenkins Action          |
+-----------------------------+----------------------+
| unlock_keychain             | true                 |
| keychain_password           | ********             |
| force                       | true                 |
| add_keychain_to_search_list | replace              |
| set_default_keychain        | true                 |
| keychain_path               | iOSKeychain.keychain |
| set_code_signing_identity   | true                 |
| output_directory            | fastlane/builds      |
| derived_data_path           | ./derivedData        |
| result_bundle               | false                |
+-----------------------------+----------------------+

[13:31:34]: Unlocking keychain: "iOSKeychain.keychain".
[13:31:34]: Unlocking keychain at path: /Users/pipedrive/Library/Keychains/iOSKeychain.keychain-db
[13:31:34]: Set output directory path to: "/path/jenkins_workspace/deploy_testflight/fastlane/builds".
[13:31:34]: Set derived data path to: "/path/jenkins_workspace/deploy_testflight/derivedData".

[13:31:34]: --------------------------------
[13:31:34]: --- Step: get_version_number ---
[13:31:34]: --------------------------------
[13:31:34]: $ cd /path/jenkins_workspace/deploy_testflight && agvtool what-marketing-version -terse
[13:31:35]: ▸ "Pipedrive.xcodeproj/../Pipedrive/Supporting Files/Info.plist"=9.0
[13:31:35]: Building with version 9.0-636
[13:31:35]: -------------------------------
[13:31:35]: --- Step: sync_code_signing ---
[13:31:35]: -------------------------------
[13:31:35]: Successfully loaded '/path/jenkins_workspace/deploy_testflight/fastlane/Matchfile' 📄

... Bunch of match stuff

[13:31:39]: All required keys, certificates and provisioning profiles are installed 🙌
[13:31:39]: Setting Provisioning Profile type to 'app-store'
[13:31:39]: -----------------------
[13:31:39]: --- Step: build_app ---
[13:31:39]: -----------------------
[13:31:39]: Successfully loaded '/path/jenkins_workspace/deploy_testflight/fastlane/Gymfile' 📄

+----------------------+-------------------------------------------------+
|               Detected Values from './fastlane/Gymfile'                |
+----------------------+-------------------------------------------------+
| workspace            | Pipedrive.xcworkspace                           |
| scheme               | Pipedrive-CI                                    |
| clean                | true                                            |
| include_bitcode      | true                                            |
| sdk                  | iphoneos11.1                                    |
| codesigning_identity | iPhone Distribution: Pipedrive Inc (ABC978YUDD) |
| configuration        | Beta                                            |
| output_name          | Pipedrive-beta.ipa                              |
+----------------------+-------------------------------------------------+

[13:31:39]: Successfully loaded '/path/jenkins_workspace/deploy_testflight/fastlane/Gymfile' 📄

+----------------------+-------------------------------------------------+
|               Detected Values from './fastlane/Gymfile'                |
+----------------------+-------------------------------------------------+
| workspace            | Pipedrive.xcworkspace                           |
| scheme               | Pipedrive-CI                                    |
| clean                | true                                            |
| include_bitcode      | true                                            |
| sdk                  | iphoneos11.1                                    |
| codesigning_identity | iPhone Distribution: Pipedrive Inc (ABC978YUDD) |
| configuration        | Beta                                            |
| output_name          | Pipedrive-beta.ipa                              |
+----------------------+-------------------------------------------------+

[13:31:39]: $ xcodebuild -list -workspace Pipedrive.xcworkspace -configuration Beta
[13:31:41]: $ xcodebuild -showBuildSettings -workspace Pipedrive.xcworkspace -scheme Pipedrive-CI -configuration Beta
+------------------------------------------------------------------------------+-----------------------------------------------------------------------------+
|                                                                   Summary for gym 2.68.2                                                                   |
+------------------------------------------------------------------------------+-----------------------------------------------------------------------------+
| export_method                                                                | app-store                                                                   |
| export_options.provisioningProfiles.com.pipedrive.dev.Pipedrive              | match AppStore com.pipedrive.dev.Pipedrive                                  |
| destination                                                                  | generic/platform=iOS                                                        |
| workspace                                                                    | Pipedrive.xcworkspace                                                       |
| scheme                                                                       | Pipedrive-CI                                                                |
| clean                                                                        | true                                                                        |
| output_directory                                                             | /path/jenkins_workspace/deploy_testflight/fastlane/builds |
| output_name                                                                  | Pipedrive-beta                                                              |
| configuration                                                                | Beta                                                                        |
| silent                                                                       | false                                                                       |
| codesigning_identity                                                         | iPhone Distribution: Pipedrive Inc (ABC978YUDD)                             |
| skip_package_ipa                                                             | false                                                                       |
| include_bitcode                                                              | true                                                                        |
| build_path                                                                   | /path/jenkins_workspace/deploy_testflight/fastlane/builds |
| derived_data_path                                                            | /path/jenkins_workspace/deploy_testflight/derivedData     |
| buildlog_path                                                                | ~/Library/Logs/gym                                                          |
| sdk                                                                          | iphoneos11.1                                                                |
| skip_profile_detection                                                       | false                                                                       |
| xcode_path                                                                   | /Applications/Xcode.app                                                     |
+------------------------------------------------------------------------------+-----------------------------------------------------------------------------+

[13:31:47]: $ set -o pipefail && xcodebuild -workspace Pipedrive.xcworkspace -scheme Pipedrive-CI -configuration Beta -sdk 'iphoneos11.1' -destination 'generic/platform=iOS' -archivePath /path/jenkins_workspace/deploy_testflight/fastlane/builds/Pipedrive-beta\ 2017-12-11\ 13.31.47.xcarchive -derivedDataPath '/path/jenkins_workspace/deploy_testflight/derivedData' clean archive CODE_SIGN_IDENTITY=iPhone\ Distribution:\ Pipedrive\ Inc\ \(ABC978YUDD\) | tee /Users/pipedrive/Library/Logs/gym/Pipedrive-Pipedrive-CI.log | xcpretty

... Lots of compiling

[13:38:42]: Successfully exported and compressed dSYM file
[13:38:42]: Successfully exported and signed the ipa file:
[13:38:42]: /path/jenkins_workspace/deploy_testflight/fastlane/builds/Pipedrive-beta.ipa
[13:38:42]: ------------------------
[13:38:42]: --- Step: testflight ---
[13:38:42]: ------------------------
[13:38:42]: Login to iTunes Connect (mail@host.com)
[13:38:45]: Login successful
[13:38:46]: Ready to upload new build to TestFlight (App: 234543564)...
[13:38:47]: Going to upload updated app to iTunes Connect
[13:38:47]: This might take a few minutes. Please don't interrupt the script.
[13:39:35]: iTunes Transporter successfully finished its job
[13:39:35]: --------------------------------------------------------------------
[13:39:35]: Successfully uploaded package to iTunes Connect. It might take a few minutes until it's visible online.
[13:39:35]: --------------------------------------------------------------------
[13:39:35]: Successfully uploaded the new binary to iTunes Connect
[13:39:35]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
[13:40:05]: Started watching build 9.0 - 634 but expected 9.0 - 636
[13:40:05]: Successfully finished processing the build 9.0 - 634
[13:40:05]: Uploaded app 9.0 - 636, but received build 9.0 - 634. If you want to wait for uploaded build to be finished processing, use the `wait_for_uploaded_build` option
[13:40:09]: Successfully set the changelog for build
[13:40:09]: Distributing new build to testers: 9.0 - 634
[13:40:09]: Successfully distributed build to Internal testers 🚀
[13:40:09]: Start download dSYM for bundle_identifier - com.pipedrive.beta.Pipedrive   version - 9.0(636)
[13:40:09]: ----------------------------
[13:40:09]: --- Step: download_dsyms ---
[13:40:09]: ----------------------------
[13:40:09]: Login to iTunes Connect (mail@host.com)
[13:40:11]: Login successful
[13:40:11]: Looking for dSYM files for com.pipedrive.beta.Pipedrive v9.0 (636)
[13:40:26]: No dSYM files found on iTunes Connect - this usually happens when no recompling happened yet
[13:40:26]: -------------------------------------------
[13:40:26]: --- Step: upload_symbols_to_crashlytics ---
[13:40:26]: -------------------------------------------
[13:40:26]: Error setting value '/path/jenkins_workspace/deploy_testflight/com.pipedrive.beta.Pipedrive-9.0-636.dSYM.zip' for option 'dsym_path'
[13:40:26]: You passed invalid parameters to 'upload_symbols_to_crashlytics'.
[13:40:26]: Check out the error below and available options by running `fastlane action upload_symbols_to_crashlytics`
+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                                                                                                                                                                                                                                                                                                                                                       Lane Context                                                                                                                                                                                                                                                                                                                                                                                                                                       |
+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM                   | ios                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| PLATFORM_NAME                      | ios                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| LANE_NAME                          | ios beta                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
| ORIGINAL_DEFAULT_KEYCHAIN          | "/Users/pipedrive/Library/Keychains/iOSKeychain.keychain-db"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
| VERSION_NUMBER                     | 9.0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| BUILD_NUMBER                       | 636                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| SIGH_PROFILE_TYPE                  | app-store                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
| IPA_OUTPUT_PATH                    | /path/jenkins_workspace/deploy_testflight/fastlane/builds/Pipedrive-beta.ipa                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
| DSYM_OUTPUT_PATH                   | /path/jenkins_workspace/deploy_testflight/fastlane/builds/Pipedrive-beta.app.dSYM.zip                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
| XCODEBUILD_ARCHIVE                 | /path/jenkins_workspace/deploy_testflight/fastlane/builds/Pipedrive-beta 2017-12-11 13.31.47.xcarchive                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |                                                                                                                                                                                                                                                                                  |
+------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
[13:40:27]: Couldn't find file at path '/path/jenkins_workspace/deploy_testflight/com.pipedrive.beta.Pipedrive-9.0-636.dSYM.zip'

+------+-------------------------------+-------------+
|                  fastlane summary                  |
+------+-------------------------------+-------------+
| Step | Action                        | Time (in s) |
+------+-------------------------------+-------------+
| 1    | default_platform              | 0           |
| 2    | setup_jenkins                 | 0           |
| 3    | is_ci                         | 0           |
| 4    | reset_git_repo                | 0           |
| 5    | get_version_number            | 0           |
| 6    | sync_code_signing             | 4           |
| 7    | build_app                     | 423         |
| 8    | changelog_from_git_commits    | 0           |
| 9    | testflight                    | 86          |
| 10   | download_dsyms                | 17          |
| 💥   | upload_symbols_to_crashlytics | 0           |
+------+-------------------------------+-------------+

[13:40:27]: fastlane finished with errors

[!] Couldn't find file at path '/path/jenkins_workspace/deploy_testflight/com.pipedrive.beta.Pipedrive-9.0-636.dSYM.zip'
Build step 'Execute shell' marked build as failure

Finished: FAILURE
```
</details>

**Comment by allu22** [comment]
@agordeev & @coverback, as fastlane's default (old) implementation uses any latest build from TestFlight, you need to use `wait_for_uploaded_build:true` with **pilot**. This way fastlane reads from ipa file what version + build you uploaded and waits for this specific build to finish processing, before continuing.

**Comment by coverback** [comment]
@allu22 Sure, I did, in `Deliverfile`, and output is already with it set. Or is it ignored from `Deliverfile`?

**Comment by allu22** [comment]
@coverback, no **pilot** does not use `Deliverfile`. You have to pass it in here: 

https://github.com/pipedrive/ios-fastlane-config/blob/master/fastlane/Fastfile#L56

**Comment by coverback** [comment]
@allu22 Thanks! You opened my eyes :D 

It always was confusing to me how it worked with Deliverfile "kinda" used, but now I found [this code](https://github.com/fastlane/fastlane/blob/9e0e66f81cc90e09130f078121e0576fc8576248/fastlane_core/lib/fastlane_core/ios_app_identifier_guesser.rb#L45) and it all makes sense.

Also very lovely of you to point it out in my repo 😂 

Seems to work for me now!
